examples/llm/src/models/mosaic_gpt/mosaic_gpt.py

```diff
@@ -64,14 +70,19 @@ def __init__(self, config: MosaicGPTConfig):
                                  device=config.init_device)
             })
         self.transformer.update({'emb_drop': nn.Dropout(config.emb_pdrop)})
-        self.transformer.update({
-            'blocks':
-                nn.ModuleList([
-                    gpt_blocks.GPTBlock(device=config.init_device,
-                                        **config.to_dict())
-                    for _ in range(config.n_layers)
-                ])
-        })
+        layers = nn.ModuleList()
+        for _ in range(config.n_layers):
+            block = gpt_blocks.GPTBlock(device=config.init_device, **config.to_dict())
+            if config.tensor_parallel_mlp and is_torch_2_or_higher() and enable_2d_with_fsdp():
+                device_type = 'cuda' if get_device(None).name == 'gpu' else 'cpu'
+                node_count = config.world_size // config.local_world_size
+                twod_mesh = DeviceMesh(device_type=device_type, mesh=torch.arange(0, config.world_size).view(node_count, -1))
```

Comment:
I wonder if we should be creating `DeviceMesh` just once, and then passing to each block?

Reply:
Let me rearrange the code. It will take care of this.